### ISSUE TITLE
Block gallery styles erroneously get output as separate style from amp-custom in Classic mode

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#1780]
Reported in https://wordpress.org/support/topic/amp-custom-is-missing-in-the-class-amp-gallery-embed-php/

On a site with Classic mode, the template is currently generating the following invalid markup in the head when a gallery is used:

```html
<style amp-custom>
...
</style>
<style>
	.wp-block-gallery.is-cropped .blocks-gallery-item amp-img > img {
		object-fit: cover;
	}
</style>
```

This seems to be due to the use of `wp_print_styles` action introduced in https://github.com/ampproject/amp-wp/pull/1627 via https://github.com/ampproject/amp-wp/commit/6b7d305ced47864ecf0c898338889e4857da7cc3.

**Comment by westonruter** [comment]
This seems to be a plugin compatibility problem with NextGEN Gallery. In particular, there is a conflict between how that plugin (for some reason) does output buffering in its resource manger component. Here is a workaround compatibility plugin: https://gist.github.com/westonruter/99b1ca0d593f44888401a396d991e3a6

See also https://wordpress.org/support/topic/amp-custom-is-missing-in-the-class-amp-gallery-embed-php/#post-11092401

**Comment by jackmcconnell** [comment]
I am also experiencing this but am not using NextGen gallery. I'm using a custom theme and have disabled all functions to try and prevent the issue from occurring but it still appears. This leads me to believe that a core WordPress function relating to images that is used by the theme may be causing the issue. Additionally, I believe better handling of when this style is output on the page should be added as, whilst using the latest version of WordPress, the whole site is using the Classic Editor plugin and as this related to a Gallery Block in the Block/Gutenberg editor, it shouldn't be being added to the page at all.

**Comment by westonruter** [comment]
I believe this issue will be made obsolete by #2202 since the entire Reader mode template will be sent through the AMP processor.

Here's a recent development build you can test to confirm: [amp.zip](https://github.com/ampproject/amp-wp/files/4032667/amp.zip) (1.5.0-alpha-20200107T191925Z-833fbf594)

**Comment by jackmcconnell** [comment]
Thanks @westonruter. Have installed and am testing but nothing in between the `if (have_posts()) : while (have_posts()) : the_post();` and `endwhile; else : endif;` is being rendered on the page. Is this expected?

**Comment by westonruter** [comment]
No, that is not expected.

**Comment by jackmcconnell** [comment]
@westonruter I appreciate this probably isn't the place for this but when using any custom theme i've tried and using a custom AMP template, nothing is rendered between the `if (have_posts()) : while (have_posts()) : the_post();` and `endwhile; else : endif;` when using Reader mode and only Website and Posts are checked in AMP > General. Reverting back to v1.4.2 renders everything as expected.

**Comment by jackmcconnell** [comment]
Ok, it seems that using the_content(); outside of the loop when on an AMP template is the fix for now. The loop seems to fail completely in this scenario. Will investigate more if I have the time.

**Comment by westonruter** [comment]
What template mode are you using? Standard, Transitional, or Reader?

So you have AMP-specific templates in your theme (in an `amp/` subdirectory)?

**Comment by jackmcconnell** [comment]
Reader Mode. I've using the `amp_post_template_file` filter to set my own template for posts (a modified single.php file).

**Comment by westonruter** [comment]
Can you share the original `single.php` file before you made changes to it?

**Comment by jackmcconnell** [comment]
Sure. See attached. 
[single.txt](https://github.com/ampproject/amp-wp/files/4039726/single.txt)

**Comment by westonruter** [comment]
@jackmcconnell Thanks very much. You identified a bug. I've opened a fix in #4185. In the future you'll be able to use The Loop as normal. Here's a new build to test: https://github.com/ampproject/amp-wp/pull/4185#issuecomment-578596406.
